polyfill/lib/casts.mjs

```diff
@@ -44,37 +45,13 @@ export function CastAbsolute(arg, aux) {
 }
 
 export function CastDateTime(arg, aux) {
+  if ('string' === typeof arg) {
+    return DateTime.fromString(arg);
+  }
   const DateTime = ES.GetIntrinsic('%Temporal.DateTime%');
-  if (HasSlot(arg, YEAR, MONTH, DAY, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND, NANOSECOND)) {
+  if (ES.IsDateTime(arg)) {
     return arg;
   }
-  if (HasSlot(arg, EPOCHNANOSECONDS)) return arg.inTimeZone(aux);
-  if (HasSlot(arg, YEAR, MONTH, DAY) && HasSlot(aux, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND, NANOSECOND)) {
-    return new DateTime(
-      GetSlot(arg, YEAR),
-      GetSlot(arg, MONTH),
-      GetSlot(arg, DAY),
-      GetSlot(aux, HOUR),
-      GetSlot(aux, MINUTE),
-      GetSlot(aux, SECOND),
-      GetSlot(aux, MILLISECOND),
-      GetSlot(aux, MICROSECOND),
-      GetSlot(aux, NANOSECOND)
-    );
-  }
-  if (HasSlot(aux, YEAR, MONTH, DAY) && HasSlot(arg, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND, NANOSECOND)) {
-    return new DateTime(
-      GetSlot(aux, YEAR),
-      GetSlot(aux, MONTH),
-      GetSlot(aux, DAY),
-      GetSlot(arg, HOUR),
-      GetSlot(arg, MINUTE),
-      GetSlot(arg, SECOND),
-      GetSlot(arg, MILLISECOND),
-      GetSlot(arg, MICROSECOND),
-      GetSlot(arg, NANOSECOND)
-    );
-  }
```

Comment:
There's only one argument in [the docs](https://github.com/tc39/proposal-temporal/blob/cefd79ca974e097c993dc580c1da45e11dc59d9c/docs/datetime.md#temporaldatetimefromthing-string--object--temporaldatetime), and there isn't a way to pass a second argument from most call sites.

It's also very easy to use `date.withTime(time)` rather than `DateTime.from(date, time)`, so I didn't think it was essential to add.

Reply:
Fair enough, thanks.